controllers/nstemplateset/nstemplatetier_test.go

```diff
@@ -209,3 +253,408 @@ func assertThatTierTemplateIsSameAs(t *testing.T, expected *toolchainv1alpha1.Ti
+	templates     []string
+	staticParams  []toolchainv1alpha1.Parameter
+	runtimeParams map[string]string
+	filters       []template.FilterFunc
+	expectedCount int
+	expectedError string
+	validate      func(t *testing.T, objects []runtimeclient.Object)
+}
+
+// Helper to create a TierTemplateRevision for testing
+func createTestTTR(name string, templates []string, params []toolchainv1alpha1.Parameter) *toolchainv1alpha1.TierTemplateRevision {
+	templateObjects := make([]runtime.RawExtension, len(templates))
+	for i, tmpl := range templates {
+		templateObjects[i] = runtime.RawExtension{Raw: []byte(tmpl)}
+	}
+
+	return &toolchainv1alpha1.TierTemplateRevision{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      name,
+			Namespace: "toolchain-host-operator",
+		},
+		Spec: toolchainv1alpha1.TierTemplateRevisionSpec{
+			TemplateObjects: templateObjects,
+			Parameters:      params,
+		},
+	}
+}
+
+// Helper to create a tierTemplate with TTR
+func createTestTierTemplate(ttr *toolchainv1alpha1.TierTemplateRevision) *tierTemplate {
+	return &tierTemplate{
+		templateRef: "test-template",
+		tierName:    "basic",
+		typeName:    "dev",
+		ttr:         ttr,
+	}
+}
+
+// Template constants to avoid duplication
+const (
+	configMapTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME}}",
+			"namespace": "{{.NAMESPACE}}"
+		},
+		"data": {
+			"config": "{{.CONFIG_VALUE}}"
+		}
+	}`
+	// #nosec G101  // This is a test template, not a real secret and does not have any sensitive data or static values passed
+	secTemplate = `{
+		"apiVersion": "v1",
+		"kind": "Secret",
+		"metadata": {
+			"name": "secret-{{.SPACE_NAME}}",
+			"namespace": "{{.NAMESPACE}}"
+		},
+		"data": {
+			"quota": "{{.QUOTA_LIMIT}}",
+			"runtime-param": "{{.RUNTIME_PARAM}}"
+		}
+	}`
+
+	namespaceTemplate = `{
+		"apiVersion": "v1",
+		"kind": "Namespace",
+		"metadata": {
+			"name": "ns-{{.SPACE_NAME}}"
+		}
+	}`
+
+	invalidTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME",
+			"namespace": "default"
+		}
+	}`
+
+	invalidFunctionTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME | invalid_function}}",
+			"namespace": "default"
+		}
+	}`
+
+	invalidJSONTemplate = `{
+		"apiVersion": "v1",
+		"kind": "ConfigMap",
+		"metadata": {
+			"name": "config-{{.SPACE_NAME}}",
+			"namespace": "default"
+		},
+		"data": {
+			"invalid": "json"
+		}
+	` // Missing closing brace
+)
+
+func TestProcessGoTemplates(t *testing.T) {
+	// Setup scheme
+	s := runtime.NewScheme()
+	err := apis.AddToScheme(s)
+	require.NoError(t, err)
+
+	// Standard runtime parameters for most tests
+	standardRuntimeParams := map[string]string{
+		"SPACE_NAME": "johnsmith",
+	}
+
+	// Table-driven tests for common scenarios
+	testCases := []templateTestCase{
+		{
+			name:      "successful processing with valid templates",
```

Comment:
I'm wondering if `TestProcessWithTTR` is a duplication of this subtest . WDYT?

Reply:
Not exactly though, but can be removed.